javaee8/jaxrs-2.1/src/main/java/org/wildfly/swarm/ts/javaee8/jaxrs/sse/SseBroadcastResource.java

```diff
@@ -0,0 +1,62 @@
+package org.wildfly.swarm.ts.javaee8.jaxrs.sse;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.sse.OutboundSseEvent;
+import javax.ws.rs.sse.Sse;
+import javax.ws.rs.sse.SseBroadcaster;
+import javax.ws.rs.sse.SseEventSink;
+
+@Path("/broadcasting")
+public class SseBroadcastResource {
+    private Sse sse;
+    private SseBroadcaster broadcaster;
+
+    public SseBroadcastResource(@Context final Sse sse) {
+        this.sse = sse;
+        this.broadcaster = sse.newBroadcaster();
+    }
+
+    @GET
+    @Path("/register")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void register(@Context SseEventSink eventSink) {
+        try (SseEventSink sink = eventSink) {
+            sink.send(sse.newEvent("Hello World"));
+            this.broadcaster.register(sink);
+            sink.send(sse.newEvent("You are registered!"));
+        }
+    }
+
+    @GET
+    @Path("/broadcast")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void broadcast(@Context SseEventSink eventSink) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        broadcaster.register(eventSink);
+        latch.countDown();
+
+        if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
Actually, what is the latch doing here? I mean, what's the difference between

```java
        CountDownLatch latch = new CountDownLatch(1);
        broadcaster.register(eventSink);
        latch.countDown();
```

and simple

```java
        broadcaster.register(eventSink);
```

?

Reply:
Right, It stayed there after I tested broadcast with more than one client. In this code, it is not needed. I will delete it.